packages/lodestar/src/db/api/beacon/repositories/block.ts

```diff
@@ -1,72 +1,46 @@
 import {Slot, SignedBeaconBlock} from "@chainsafe/lodestar-types";
 import {IBeaconConfig} from "@chainsafe/lodestar-config";
 
-import {BulkRepository} from "../repository";
-import {ChainRepository} from "./chain";
 import {IDatabaseController} from "../../../controller";
-import {Bucket, encodeKey} from "../../../schema";
+import {Bucket, encodeKey} from "../../schema";
+import {Repository} from "./abstract";
+import {ChainRepository} from "./chain";
 
-export class BlockRepository extends BulkRepository<SignedBeaconBlock> {
+/**
+ * Blocks by root
+ *
+ * Used to store unfinalized blocks
+ */
+export class BlockRepository extends Repository<Uint8Array, SignedBeaconBlock> {
 
   private chain: ChainRepository;
 
   public constructor(
     config: IBeaconConfig,
-    db: IDatabaseController,
+    db: IDatabaseController<Buffer, Buffer>,
     chain: ChainRepository) {
     super(config, db, Bucket.block, config.types.SignedBeaconBlock);
     this.chain = chain;
   }
 
+  /**
+   * Id is hashTreeRoot of unsigned BeaconBlock
+   */
   public getId(value: SignedBeaconBlock): Uint8Array {
     return this.config.types.BeaconBlock.hashTreeRoot(value.message);
   }
 
-  public async set(id: Uint8Array, value: SignedBeaconBlock): Promise<void> {
+  public async put(id: Uint8Array, value: SignedBeaconBlock): Promise<void> {
     await Promise.all([
-      this.db.put(encodeKey(Bucket.blockSlotRefs, value.message.slot), id),
-      this.db.put(encodeKey(Bucket.blockRootRefs, id), this.config.types.Slot.serialize(value.message.slot)),
-      super.set(id, value)
+      this.db.put(encodeKey(Bucket.blockSlotRefs, value.message.slot), id as Buffer),
+      super.put(id, value),
     ]);
   }
-
-  public async getFinalizedBlock(): Promise<SignedBeaconBlock | null> {
-    const root = await this.chain.getFinalizedBlockRoot();
-    if(!root) return null;
-    return await this.get(root);
-  }
-
-  public async getJustifiedBlock(): Promise<SignedBeaconBlock | null> {
-    const root = await this.chain.getJustifiedBlockRoot();
-    if(!root) return null;
-    return await this.get(root);
-  }
-
-  public async getBlockBySlot(slot: Slot): Promise<SignedBeaconBlock | null> {
+  public async getBySlot(slot: Slot): Promise<SignedBeaconBlock | null> {
```

Comment:
This should return array of SignedBeaconBlock, only in archive we have 1-1 mapping between slot and root

Reply:
agreed that this is not right, will tackle as part of another PR, when thinking about https://github.com/ChainSafe/lodestar/issues/724